R/pkg/inst/worker/worker.R

```diff
@@ -100,7 +104,20 @@ if (isEmpty != 0) {
     # Timing reading input data for execution
     inputElap <- elapsedSecs()
 
-    output <- computeFunc(partition, data)
+    if (isDataFrame) {
+      if (deserializer == "row") {
+        # Transform the list of rows into a data.frame
+        data <- do.call(rbind.data.frame, c(data, stringsAsFactors = FALSE))
+      }
```

Comment:
If deserializer is not `row` can we add a check to see if `data` is a valid `data.frame` ? (I'm guess the UDFs assume that is the input type)